File: maximumPathSumI/018.py
test_triangle = [[1, 0, 0, 0],
                [2, 10, 0, 0],
                [6, 5, 10, 0],
                [50, 5, 6, 10]
]

def maximumPathSumI(test_triangle):
    for i in range(len(test_triangle) - 1, -1, -1):
        for j in range(i):
    		test_triangle[i - 1][j] += max(test_triangle[i][j], test_triangle[i][j+1])
    return test_triangle[0][0]

print(maximumPathSumI(test_triangle))

File: maximumPathSumI/018.js
const testTriangle = [
	[3, 0, 0, 0],
	[7, 4, 0, 0],
	[2, 4, 6, 0],
	[8, 5, 9, 3],
];

function maximumPathSumI(testTriangle) {
	for (let i = testTriangle.length - 1; i >= 0; i--) {
		for (let j = 0; j < i; j++) {
			testTriangle[i - 1][j] += Math.max(testTriangle[i][j], testTriangle[i][j + 1]);
		}
	}
	return testTriangle[0][0];
}

console.log(maximumPathSumI(testTriangle));
